File: backend/app/schemas/clinician.py
from pydantic import BaseModel

from .brief import DivergenceMoment


class TranscriptItem(BaseModel):
    timestamp: str
    speaker: str
    text: str
    affect: str


class TrendItem(BaseModel):
    label: str
    direction: str


class PatientBrief(BaseModel):
    patient_id: str
    patient_name: str
    assigned_clinician: str
    next_appointment: str
    last_checkin_label: str
    risk_level: str
    trend_label: str
    summary: str
    clinical_summary: str
    what_changed: str
    key_themes: list[str]
    opening_questions: list[str]
    divergence_moments: list[DivergenceMoment]
    transcript: list[TranscriptItem]
    trends: list[TrendItem]

File: frontend/types/clinician.ts
import type { DivergenceMoment, RiskLevel } from "@/types/brief";

export interface TranscriptItem {
  timestamp: string;
  speaker: "P" | "C";
  text: string;
  affect: string;
}

export interface TrendItem {
  label: string;
  direction: "up" | "stable" | "down";
}

export interface PatientBrief {
  patient_id: string;
  patient_name: string;
  assigned_clinician: string;
  next_appointment: string;
  last_checkin_label: string;
  risk_level: RiskLevel;
  trend_label: string;
  summary: string;
  clinical_summary: string;
  what_changed: string;
  key_themes: string[];
  opening_questions: string[];
  divergence_moments: DivergenceMoment[];
  transcript: TranscriptItem[];
  trends: TrendItem[];
}
